src/App.tsx

```diff
@@ -1,24 +1,252 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { NewFormTodo } from './components/NewFormTodo';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Notifications } from './components/Notificiations';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodo,
+  removeTodo,
+  changeTodo,
+} from './api/todos';
+import { Filter } from './types/FilterBy';
+import { filterTodos } from './utils/filterTodos';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+const USER_ID = 6414;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState <Error>(Error.NONE);
+  const [filterType, setFilterType] = useState<Filter>(Filter.ALL);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isAddWaiting, setIsAddWaiting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleteWaiting, setIsDeleteWaiting] = useState(false);
+  const [todosIdsToRemove, setTodosIdsToRemove] = useState<number[]>([]);
+  const [todosIdsToUpdate, setTodosIdsToUpdate] = useState<number[]>([]);
+  const [isUpdateWaiting, setIsUpdateWaiting] = useState(false);
```

Comment:
can we use one variable for waiting some response from server instead of creating separate boolean variable for each action?